include/effects/matrix/PatternWeather.h

```diff
@@ -250,8 +253,35 @@ class PatternWeather : public LEDStripEffect
 
 public:
 
-    PatternWeather() : LEDStripEffect("Weather")
+    PatternWeather() : LEDStripEffect(EFFECT_MATRIX_WEATHER, "Weather")
     {
+        strPostalCode = cszZipCode;
+        strCountryCode = cszCountryCode;
+    }
+
+    PatternWeather(const JsonObjectConst&  jsonObject) : LEDStripEffect(jsonObject)
+    {
+        strPostalCode = jsonObject["pcd"].as<String>();
+        strCountryCode = jsonObject["ccd"].as<String>();
+
+        if (strPostalCode.isEmpty())
+            strPostalCode = cszZipCode;
+
+        if (strCountryCode.isEmpty())
+            strCountryCode = cszCountryCode;
+    }
+
+    virtual bool SerializeToJSON(JsonObject& jsonObject) 
+    {
+        StaticJsonDocument<128> jsonDoc;

```

Comment:
Will 128 always be large enough to serialize it here?

Reply:
That's a fair question. What needs to be serialized is:

* Two property names of 2 characters each (6 bytes)
* Two property names of 3 characters each (8 bytes)
* One integer (8 bytes)
* "Weather" (8 bytes)
* One country code of 2 characters (3 bytes)
* The postal code or other location indicator

The knowns add up to 33 bytes in raw data, leaving 95 bytes for postal code/location and bytes for structure. I gathered that should be enough.

Now that you asked the question I've used the ArduinoJson Assistant, and it calculated that 82 bytes are needed for the serialization with my location data (postal code "4133EK", country code "nl"). That leaves 46 more bytes (&asymp; characters) for a different location specifier. I still think that's enough. :)